src/doc/en/developer/coding_basics.rst

```diff
@@ -1179,21 +1236,55 @@ framework. Here is a comprehensive list:
             sage: 2+2  # known bug
             5
 
+  - **modularization:** To enable
+    :ref:`separate testing of the distribution packages <section-doctesting-venv>`
+    of the modularized Sage library, doctests that depend on features provided
+    by other distribution packages can be tagged ``# needs FEATURE``.
+    For example:
+
+    .. CODE-BLOCK:: rest
+
+        Consider the following calculation::
+
+            sage: a = AA(2).sqrt()  # needs sage.rings.number_field
+            sage: b = sqrt(3)       # needs sage.symbolic
+            sage: a + AA(b)         # needs sage.rings.number_field sage.symbolic
+            3.146264369941973?
+
   .. NOTE::
 
-      - Any words after ``# optional`` are interpreted as a list of
+      - Any words after ``# optional`` and ``# needs``  are interpreted as a list of
         package (spkg) names or other feature tags, separated by spaces.
 
       - Any punctuation other than underscores (``_``) and periods (``.``),
         that is, commas, hyphens, semicolons, ..., after the
         first word ends the list of packages.  Hyphens or colons between the
         word ``optional`` and the first package name are allowed.  Therefore,
-        you should not write ``optional: needs package CHomP`` but simply
-        ``optional: CHomP``.
+        you should not write ``# optional - depends on package CHomP`` but simply
+        ``# optional - CHomP``.
 
-      - Optional tags are case-insensitive, so you could also write ``optional:
+      - Optional tags are case-insensitive, so you could also write ``# optional -
         chOMP``.
 
+  If ``# optional`` or ``# needs`` is placed right after the ``sage:`` prompt,
+  it is a codeblock-scoped tag, which applies to all doctest lines until
+  a blank line is encountered.
+
```

Comment:
As for "codeblock-scoped tag", the initial word "code" seems redundant  since the tags are already used within code block. The term "block-scoped tag" is already long, so why make it longer? I prefer "block-scoped tag".

On the other hand, the term "persistent tag" that I originally used is shorter but somewhat obscure...

I will not insist if you disagree though.

Reply:
"block-scoped" is fine with me, I'll change it

Reply:
Fixed in c6d1b89

Reply:
Thanks.